dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/DatasetServiceTest.java

```diff
@@ -75,4 +92,61 @@ public void removeDatasetThumbnail() {
         //then
         verify(commandEngine, times(1)).submit(any(UpdateDatasetThumbnailCommand.class));
     }
+
+    @Test
+    public void setDatasetEmbargoDate() {
+        // given
+        Dataset dataset = new Dataset();
+        Date embargoDate = Date.from(Instant.now().truncatedTo(ChronoUnit.DAYS).plus(2, ChronoUnit.DAYS));
+        when(datasetDao.merge(dataset)).thenReturn(dataset);
+
+        // when
+        Dataset result = datasetService.setDatasetEmbargoDate(dataset, embargoDate);
+
+        // then
+        Assertions.assertEquals(embargoDate, result.getEmbargoDate().getOrNull());
+    }
+
+    @Test
+    public void setDatasetEmbargoDate_lockedDataset() {
+        // given
+        Dataset dataset = new Dataset();
+        DatasetLock lock = new DatasetLock(DatasetLock.Reason.InReview, MocksFactory.makeAuthenticatedUser("Jurek", "Kiler"));
+        lock.setId(1L);
+        dataset.addLock(lock);
+        Date embargoDate = Date.from(Instant.now().truncatedTo(ChronoUnit.DAYS).plus(2, ChronoUnit.DAYS));
+
+        // when
+        Exception exception = Assertions.assertThrows(IllegalStateException.class, () -> {
+            datasetService.setDatasetEmbargoDate(dataset, embargoDate);
+        });
+
+        // then
+        String message = "Update embargo date failed. Dataset is locked. [DatasetLock[ id=1 ]]";
+        Assertions.assertEquals(message, exception.getMessage());
```

Comment:
In this case we should either compare only invariant part of the message: extract it to the constant in the place where it's defined and use this constant also here or create a method that produces message for exception and call it within test to produce expected message. 
Currently any change to message or ```DatasetLock#toString()``` will break the test.